borg/helpers.py

```diff
@@ -506,6 +535,34 @@ def parse_pattern(pattern, fallback=FnmatchPattern):
     return cls(pattern)
 
 
+def parse_exclude_pattern(pattern, fallback=FnmatchPattern):
+    """Read pattern from string and return an instance of the appropriate implementation class.
+    """
+    epattern = parse_pattern(pattern, fallback)
+    return InclExclPattern(epattern, False)
+
+
+def parse_inclexcl_pattern(pattern, fallback=ShellPattern):
+    """Read pattern from string and return a InclExclPattern object."""
+    type_prefix_map = {
+        '-': False,
+        '+': True,
+        'R': RootPath,
+        'r': RootPath,
+    }
+    ptype = None
+    if len(pattern) > 1 and pattern[0] in type_prefix_map:
+        (ptype, pattern) = (type_prefix_map[pattern[0]], pattern[1:])
+        pattern = pattern.lstrip()
+    if ptype is None or not pattern:
+        raise argparse.ArgumentTypeError("Unable to parse pattern: {}".format(pattern))
```

Comment:
Perhaps writing this in [EAFP](https://docs.python.org/3/glossary.html#term-eafp)-style is easier to read? Decide for yourself.

```py
try:
    ptype = type_prefix_map[pattern[0]]
except (IndexError, KeyError):
    raise argparse.ArgumentTypeError(...)
pattern = pattern[1:].lstrip()  # below the ptype= might be clearer, although this doesn't raise.
```